DEVELOPING.md

```diff
@@ -17,7 +32,131 @@ This project:
+4. uses [secret-shield](https://github.com/mapbox/secret-shield) to help block secrets such as access tokens from being exposed. Setup `secret-shield` by running:
+```
+npm install -g @mapbox/secret-shield
+scripts/install-pre-commit/install-pre-commit.sh
+```
+
+5. uses CircleCI and AWS Device Farm for continuous integration.
+
+### Accessing the Maps SDK's source
+
+Clone the git repository:
+
+```
+git clone git@github.com:mapbox/mapbox-maps-ios.git && cd mapbox-maps-ios
+```
+
+### Building the Maps SDK
+
+Pre-requisite: Valid `.netrc` file located on your machine at `~/.netrc`. Please visit https://docs.mapbox.com/ios/maps/overview for additional information about setting up your `.netrc` file.
+
+Open `Package.swift` and build the MapboxMaps target.
+
+### Update dependency versions
+
+You must remember to update the Package.swift file in https://github.com/mapbox/mapbox-maps-ios/blob/main/Package.swift, not the App.xcworkspace, using the following commands:
+
+```
+xed Package.swift
+//update versions in there
+swift package update
+```
+
+### Running the Debug App
+
+In order to use the debug app, open Apps/Apps.xcworkspace and build and run the `DebugApp` target.
+
+You must provide a Mapbox access token to display Mapbox-hosted maps in the `DebugApp` test application. Add your Mapbox token to the `MBXAccessToken` key in `DebugApp/DebugApp/Info.plist`. Alternatively, create a [file at `~/mapbox`](https://docs.mapbox.com/help/troubleshooting/private-access-token-android-and-ios/#ios) containing your access token.
+
+### CircleCI
+
+CircleCI's [YAML](https://en.wikipedia.org/wiki/YAML) format can be picky. You can run `make validate` to ensure you've formatted the changes to the config file correctly. This make target will install the [CircleCI command line interface](https://circleci.com/docs/2.0/local-cli/) to validate `.circleci/config.yml`, and run `circleci config validate`
+
+# Testing
+
+## Unit Testing on devices
+
+To run device tests there are few options:
+
+1. Run via Xcode:
+
+  - Install [xcodegen](https://github.com/yonaskolb/XcodeGen).
+  - Run `$ xcodegen` in the root of the repo.
+  - Open the resulting `MapboxMaps.xcodeproj`.
+  - Test (Cmd-U) the `MapboxTestHost` scheme.
+
+2. Trigger via CI by adding `[run device tests]` to a git commit message. These tests also run "nightly".
+(Examples can also be run as tests on CI by adding `[run app device tests]`.)
+
+3. Trigger tests on AWS Device Farm from the command line. Note that this requires certain environment variables to be set; please see the makefile for these.
+`make test-with-device-farm SCHEME=MapboxTestHost APP_NAME=MapboxTestHost`.
+
+4. Trigger tests on a local device (connected by USB) using the same setup as Device Farm testing by running:
+`make local-test-with-device-farm-ipa SCHEME=MapboxTestHost CONFIGURATION=Release ENABLE_CODE_SIGNING=1`
```

Comment:
I'm not sure any of these are supported anymore. @OdNairy has been working on some improvements here. I know for 2, we know trigger device tests in the CircleCI web app.

Reply:
We do not support that makefile rules anymore